Framework/CurveFitting/src/Functions/ProcessBackground.cpp

```diff
@@ -415,8 +414,8 @@ void ProcessBackground::addRegion() {
   }
 
   // Check
-  for (size_t i = 1; i < vx.size(); ++i) {
-    if (vx[i] <= vx[i - 1]) {
+  for (auto it = vx.begin() + 1; it != vx.end(); ++it) {
+    if (*it <= *it - 1) {
       g_log.error()
           << "The vector X with value inserted is not ordered incrementally\n";
       throw std::runtime_error("Build new vector error!");
```

Comment:
Using the `HistogramData::isValid()` method appears to cause a significant amount of slowdown (i.e. `1.96s` rather than `1.88s`) for some reason.